contracts/SmartWalletImplementation.sol

```diff
@@ -122,12 +145,66 @@ contract SmartWalletImplementation is SmartWalletStorage, ISmartWalletImplementa
         );
     }
 
+    function defaultGetExpectedIn(address swapContract, ISwap.GetExpectedInParams memory params)
+        private
+        view
+        returns (uint256 srcAmount)
+    {
+        uint8 srcDecimal = 18;
+        if (params.tradePath[0] != address(ETH_TOKEN_ADDRESS)) {
```

Comment:
we dont have this case, as for getExpectedRate, things are required to be wrapped token.